### PR TITLE
update publish types

### PR DESCRIPTION
**Pull request by alanrenmsft**
currently, the master branch of our DT fork is not guaranteed to be up to date, add step to force it to be the same as the remote branch.

in this build below, there is an error message, and the generated PR is not mergable due to conflicts.

https://mssqltools.visualstudio.com/CrossPlatBuildScripts/_build/results?buildId=132950&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=35f53584-79fc-539f-463c-6471d0663a3a 

NOTE: I will test the changes in this PR later.

### CODE REVIEW

File: build/azure-pipelines/publish-types/publish-types.yml [hunk 1]
```diff
@@ -44,18 +44,20 @@ steps:
...
+      git fetch upstream
+      git reset --hard upstream/master
+      git push --force
+
+      # Update the type
       node build/azure-pipelines/publish-types/update-types.js
 
       TAG_VERSION=$(git describe --tags `git rev-list --tags --max-count=1`)
 
       cd DefinitelyTyped
```
Comment (kburtram) [inline build/azure-pipelines/publish-types/publish-types.yml:59]: does the `cd` need to occur before running the git commands?

**Review by kburtram — APPROVED** [review]
LGTM but there is some updates in the ordering of the commands that may change behavior.

**Comment by alanrenmsft** [comment]
mark as draft while I test the changes in a pipeline

**Review by Charles-Gagnon — APPROVED** [review]

**Comment by alanrenmsft** [comment]
here is the test build: https://mssqltools.visualstudio.com/CrossPlatBuildScripts/_build/results?buildId=134984&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=35f53584-79fc-539f-463c-6471d0663a3a

File: build/azure-pipelines/publish-types/publish-types.yml [hunk 2]
```diff
@@ -44,17 +44,20 @@ steps:
...
       # Sync up to latest from the DT repo
       git remote add upstream https://github.com/DefinitelyTyped/DefinitelyTyped.git
       git fetch upstream
-      git merge upstream/master
-      git push origin
+      git reset --hard upstream/master
+      git push --force
+
+      # Update the type
+      cd ..
```
Comment (Charles-Gagnon) [inline build/azure-pipelines/publish-types/publish-types.yml:56]: You can also just use pushd/popd

**Review by Charles-Gagnon — APPROVED** [review]
Thanks for fixing this!

**Pull request merged**